### PR TITLE
fixed: Dir cache should ignore url (protocol) options (fixes #16469)

### LINKED ISSUE #16469: Feature Request: Move library update and other processes outside of Kodi GUI exe
I would like to have library scans, etc. to be able to run as a service outside of Kodi running.  For example on Windows if I shut down Kodi, I don't want to have to do scans when I come in if they could be running separately.  

Also this would be good for Nvidia Shield where Kodi isn't always active (although I am not sure if there is something similar to a service in andoird)

### PR DESCRIPTION
**Pull request by arnova**
Followup for https://github.com/xbmc/xbmc/pull/8717

### CODE REVIEW

**Comment by arnova** [comment]
jenkins build this please

**Comment by arnova** [comment]
@mkortstiege / @Montellese : Please review/comment

File: xbmc/URL.cpp [hunk 1]
```diff
@@ -547,17 +547,39 @@ std::string CURL::Get() const
...
+
+  std::string strURL;
+  strURL.reserve(sizeneed);
+
+  strURL = GetWithoutFilename();
+  strURL += m_strFileName;
+
+  return strURL;
+}
+
```
Comment (ace20022) [inline xbmc/URL.cpp]: This seems to be equivalent to
```c++
std::string CURL::GetWithoutOptions() const
{
  if (m_strProtocol.empty())
    return m_strFileName;
  
  return GetWithoutFilename() + m_strFileName;
}
```
Or have I missed something?

Comment (arnova) [inline xbmc/URL.cpp]: I think your right but note that this is just a c&p of what was already there. Perhaps there's a reason it was done this way?

**Comment by arnova** [comment]
@ace20022 : Updated

jenkins build this please

**Comment by ace20022** [comment]
@arnova thx, I can't comment on the functional change.

**Comment by Razzeee** [comment]
For the record build is fine, just failed due to addons on Windows

**Comment by Razzeee** [comment]
@arnova can you prepare a backport?

**Comment by DaveTBlake** [comment]
Thanks for doing this @arnova, a deeper solution than my attempt.

@MartijnKaijser this is the work that needs merge into Jarvis not #8717
@Razzeee I feel like I am in a Git typing race!

**Comment by arnova** [comment]
Will try to prepare a backport for Jarvis today.

**Comment by ronie** [comment]
@arnova looks like this breaks music playback from the library.
(i can still play music from the files node)

http://pastebin.com/SmXcsmUQ

**Comment by arnova** [comment]
I'll look into it.

**Comment by arnova** [comment]
@ronie: Please test https://github.com/xbmc/xbmc/pull/8797

**Pull request merged**